app/models/user.rb

```diff
@@ -19,7 +19,19 @@ class User < ApplicationRecord
    { attr_name: :account_id, new_attr_name: :Account, value: Proc.new { |account_id| Account.find_by(id: account_id).name } }
   )
 
-  def set_default_role
-    self.role ||= self.account.find_or_create_by(name: Role::REQUESTER)
+  def role
+    if account.is_a?(LegacyAccount)
```

Comment:
Can you explain why we are doing like this?

Reply:
A user can have multiple roles via user_role, if user is of `LegacyAccount` (means a account which does not have sub-accounts) then its role is stored directly in user table, and on the other hand for multiple account case, we are using user_roles.